src/components/ReportActionItem/TaskPreview.tsx

```diff
@@ -24,78 +21,62 @@ import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
 import Navigation from '@libs/Navigation/Navigation';
 import * as ReportUtils from '@libs/ReportUtils';
 import * as TaskUtils from '@libs/TaskUtils';
-import reportActionPropTypes from '@pages/home/report/reportActionPropTypes';
 import * as Session from '@userActions/Session';
 import * as Task from '@userActions/Task';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {Policy, Report, ReportAction} from '@src/types/onyx';
+import {isNotEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    /** The ID of the associated taskReport */
-    taskReportID: PropTypes.string.isRequired,
-
-    /** Whether the task preview is hovered so we can modify its style */
-    isHovered: PropTypes.bool,
-
-    /** The linked reportAction */
-    action: PropTypes.shape(reportActionPropTypes).isRequired,
-
+type TaskPreviewOnyxProps = {
     /* Onyx Props */
 
-    taskReport: PropTypes.shape({
-        /** Title of the task */
-        reportName: PropTypes.string,
-
-        /** AccountID of the manager in this iou report */
-        managerID: PropTypes.number,
-
-        /** AccountID of the creator of this iou report */
-        ownerAccountID: PropTypes.number,
-    }),
+    taskReport: OnyxEntry<Report>;
 
     /** The policy of root parent report */
-    rootParentReportpolicy: PropTypes.shape({
-        /** The role of current user */
-        role: PropTypes.string,
-    }),
+    rootParentReportpolicy: OnyxEntry<{role: string}>;
+};
 
-    /** The chat report associated with taskReport */
-    chatReportID: PropTypes.string.isRequired,
+type TaskPreviewProps = WithCurrentUserPersonalDetailsProps &
+    TaskPreviewOnyxProps & {
+        /** The ID of the associated policy */
+        // eslint-disable-next-line
```

Comment:
why this is needed ?

Reply:
There is a lint error that shows `policyId is defined but not used anywhere` so I added it.

Reply:
A few things:

- If the prop is not used anywhere, remove it
- If the eslint-disable-next-line needs to stay, disable the specific rule instead of all rules and also add a code comment for why the prop needs to be there even though it isn't used anywhere

I believe the reason `policyID` is here is because it is used in `withOnyx` to fetch `rootParentReportpolicy`. Since ESLint won't detect it's usage in `withOnyx`, that's why it throws the error.

Reply:
Please do one of the options I posted in my previous comment, I don't see that either of them were followed.